### PR TITLE
Support ares-log

### PR DESCRIPTION
**Pull request by ssuminahn**
:Release Notes:
Support ares-log

:Detailed Notes:
ares-log provides the user with a more convenient function to check the log.
Most of the options provided by the existing journald can be utilized.
In addition, it provides a function to save log, search log file list, and search unit list.

:Testing Performed:
1. Pass unit test on ose and auto
2. Pass eslint
3. Check below options
- node ares-log.js -sd pmlogd
- node ares-log.js -cd
- node ares-log.js -f
- node ares-log.js -r
- node ares-log.js -n 7
- node ares-log.js -k
- node ares-log.js -b
- node ares-log.js -S today
- node ares-log.js -U yesterday
- node ares-log.js -S 2021-04-07 21:45:00 --until 2021-04-07 21:46:00
- node ares-log.js -f -pid 1834
- node ares-log.js -f -o json
- node ares-log.js -n 7 -s
- node ares-log.js -f -s
- node ares-log.js -s tmp.log
- node ares-log.js -s tmp
- node ares-log.js -f -o -d ose
- node ares-log.js -f -o jjson -d ose
- node ares-log.js --file-list
- node ares-log.js --file system.journal
-  node ares-log.js -ul
- node ares-log.js -ul -dp 1
- node ares-log.js -u com.webos.service.position -d auto
- node ares-log.js -u sam -dp 1 -f -d auto

:Issues Addressed:
[PLAT-134677] Support to unit test of ares-log

### CODE REVIEW

**Comment by ssuminahn** [comment]
@YoungeunKim 
This review request is in progress.
The bug or code may not be perfect yet. Please consider and review.
If you find any bugs, please leave a comment on the ongoing jira task.

File: lib/base/novacom.js [hunk 1]
```diff
@@ -919,7 +924,8 @@ const fs = require('fs'),
                     write[extended].bind(obj[extended])(data);
                 }).stderr.on('data', function(data) {
                     log.verbose('Session#run()', "on data (stderr)");
-                    write.stderr.bind(obj.stderr)(data);
+                    orgErrMsg = data.toString();
+                    // write.stderr.bind(obj.stderr)(data);
```
Comment (YoungeunKim) [inline lib/base/novacom.js:928]: Please check there is no side effect.

File: bin/ares-log.js [hunk 2]
```diff
@@ -0,0 +1,206 @@
+#!/usr/bin/env node
+
+/*
+ * Copyright (c) 2020 LG Electronics Inc.
```
Comment (YoungeunKim) [inline bin/ares-log.js:4]: 2021?

File: lib/log.js [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+                _makeSession,
+                _createLogFile,
+                function(next){
+                    options.session.run(options.cmd, process.stdin, _onData, process.stderr, next);
+                },
+            ],  function(err) {
+                next(err, null);
+            });
+
+            function _createLogFile(next) {
```
Comment (YoungeunKim) [inline lib/log.js:77]: It would be better to print the created log file name to user like ares-device capture.

**Comment by YoungeunKim** [comment]
@ssuminahn , Verified ares-log options execution and reviewed code. 
It seems that there are bugs, but I expect that you will fix it. 
If you run "eslint", it helps you code clean up.
Please add TCs and ares-log -h including several  negative cases and rebase develop branch.

Please continue your work. I'll comment for details for your activated RP.

File: bin/ares-log.js (hunk 2 above)
Comment (ssuminahn) [reply bin/ares-log.js]: I updated it.

File: lib/base/novacom.js (hunk 1 above)
Comment (ssuminahn) [reply lib/base/novacom.js]: When I checked, there was no side effect.

File: lib/log.js (hunk 3 above)
Comment (ssuminahn) [reply lib/log.js]: Sorry, I don't know what it means...
Please explain verbally.

**Comment by ssuminahn** [comment]
code :+1: 
verify :+1: 

I will update the help file of ares-log

**Review by YoungeunKim — COMMENTED** [review]
At first, please check eslint
lib/log.js
  236:13  warning  'err' is never reassigned. Use 'const' instead  prefer-const

1. You used both "pmlogd" and "pmLogd". Please check that they ares used properly.

2. I hope "-sd" is not show to help and make not to use by USER(Only journald is supported) 

3. -sd is working as expected? Please check.
$ ./ares-log.js -sd pmlogd
Switched log daemon to pmlogd
$ ./ares-log.js -cd
Current log daemon is journald

4. Need to valid deamon value. "test" is affect to result
$ ./ares-log.js -sd test
Switched log daemon to test

5. No error heading 
$ node ares-log.js -f test -d rpi4
ares-log ERR! Failed to add match 'test': Invalid argument

$ node ares-log.js -f -o jjson -d rpi4
ares-log ERR! Unknown output format 'jjson'.

-> There is no ERR header format. It should be 
ares-log ERR! [ssh exec failure/jornald failure] Failed to add match 'test': Invalid argument
ares-log ERR! [ssh exec failure/jornald failure] Unknown output format 'jjson'.
cf)
$ node ares-log.js -f test -d auto
ares-log ERR! [ssh exec failure]: Command 'journalctl --follow test' exited with code=1 (signal: undefined)

7. It would be better to show 20210507_133817.log filename to user when -save option is terminated
8. node ares-log.js -s tmp.log -> User can set to not show log print to console?

File: lib/base/novacom.js [hunk 4]
```diff
@@ -30,8 +29,12 @@ const fs = require('fs'),
     const keydir = path.resolve(process.env.HOME || process.env.USERPROFILE, '.ssh');
     let cliData;
 
-    function makeExecError(cmd, code, signal) {
+    function makeExecError(cmd, code, signal, orgErrMsg) {
         let err = null; // null:success, undefined:did-not-run, Error:failure
+
+        if(orgErrMsg)
```
[the remark this answers is not in the document]
Comment (YoungeunKim) [reply lib/base/novacom.js:35]: Please add CLI [error] heading and change error text clolor.

File: package.json [hunk 5]
```diff
@@ -72,7 +72,7 @@
     "eslint-plugin-import": "2.22.0",
     "jasmine": "3.6.1",
     "jasmine-pretty-html-reporter": "0.2.5",
-    "jasmine-spec-reporter": "4.2.1"
+    "jasmine-spec-reporter": "^4.2.1"
```
Comment (YoungeunKim) [inline package.json:75]: Please remove ^

**Review by ssuminahn — COMMENTED** [review]
I apply reviewed items. But I need more time to solve some items.
First of all I push new commit applied reviewed in pull request #31.

**Comment by ssuminahn** [comment]
This PR has been replaced by https://github.com/webosose/ares-cli/pull/31.
I close this PR.

**Pull request closed without merging**